src/components/KnowledgeGraphModal/KnowledgeGraphModal.tsx

```diff
@@ -0,0 +1,308 @@
+      setStyle({ width: 500, height: 300 });
+    }
+  }, [containerRef, domIDs.root]);
+
+  // update the width of the component whenever the parent container changes
+  React.useEffect(() => {
+    function handleResize() {
+      if (containerRef.current) {
+        setStyle({ width: 500, height: 300 });
+      }
+    }
+
+    // add an event listener to listen for resize events
+    window.addEventListener("resize", handleResize);
+  });
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const graphStyle = require("./cy-style.json");
+
+  const layout = {
+    name: "concentric",
+    startAngle: 0 * Math.PI,
+    minNodeSpacing: 20,
+    edgeLengthVal: 80,
+    concentric: function (node: any) {
+      return node.degree();
+    },
+    levelWidth: function (nodes: []) {
+      return 2;
+    },
+    fit: "true",
+    animate: "true",
+  };
+
+  graphStyle[1].style["background-color"] = (node: any) =>
+    getColorBasedOnType(node);
+
+  const [node, selectNode] = React.useState<Record<string, string> | null>(
+    getMainNode(elements),
+  );
+
+  const Container = styled.div`
+    display: flex;
+    justify-content: center;
+  `;
+
+  const InnerContainer = styled.div`
+    display: flex;
+    align-items: center;
+    padding: 0 10px; /* add padding between InnerContainer elements */
+  `;
+
+  const Cube = styled.div`
+    width: 10px;
+    height: 10px;
+    border-radius: 2px; /* round the edges */
+    background-color: #ff0000;
+    margin: 0 10px;
+  `;
+
+  const PaperInfo = styled.div`
+    paddingright: 20px;
+  `;
+
+  function zoomOut() {
+    myCy
+      .animation({
+        fit: {
+          eles: newElements,
+          padding: 10,
+        },
+        easing: "ease-in-out",
+        duration: 1000,
+      })
+      .play();
+  }
+
+  const getElements = async (paperId: string) => {
+    setNewElements([]);
+    try {
+      const result = await queryBackend("/paperKnowledgeGraph", {
+        body: {
+          paperId,
+        },
+      });
+      if (result.length > 1) {
+        setError(false);
+        selectNode(getMainNode(result));
+        setNewElements(result);
+      } else {
+        setError(true);
+      }
+    } catch (e) {
+      setError(true);
+    }
+  };
+
+  return (
+    <>
+      {isVisible ? (
+        <div
+          id={domIDs.root}
+          className={cs("sa-knowledge-graph-modal", className)}
+          data-testid={dataTestIDs.root}
+        >
+          <div
+            className={cs("sa-knowledge-graph-modal-wrapper", className)}
+            onClick={(e) => e.stopPropagation()}
+          >
+            <TitleClose titleText="Knowledge Graph" handleClose={toggle} />
+            {error && (
+              <div>
+                <h3>An error occurred during graph loading!</h3>
+              </div>
+            )}
+            {!error && (
+              <div className={cs("sa-knowledge-graph-wrapper", className)}>
+                <Container>
```

Comment:
I think we can add style={{marginBottom: "8px"}} here

Reply:
i was thinking of adding it in the <Container>, not the <div>, so there's a gap between the legend and the graph